st3/lsp_utils/_client_handler/decorator.py

```diff
@@ -9,8 +9,10 @@
     "register_decorated_handlers",
 ]
 
+T = TypeVar('T')
 # the first argument is always "self"
```

Comment:
We can annotate the first argument (`self`) as `ClientHandlerInterface`.

Reply:
That doesn't work and I don't have enough resources to figure out why is that.

Possibly due to covariance/contravariance thing.